pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/inv/text/LuceneFSTIndexCreator.java

```diff
@@ -50,36 +56,56 @@ public class LuceneFSTIndexCreator implements FSTIndexCreator {
    *
    * @param indexDir  Index directory
    * @param columnName Column name for which index is being created
+   * @param tableNameWithType table name with type
    * @param sortedEntries Sorted entries of the unique values of the column.
    * @throws IOException
    */
-  public LuceneFSTIndexCreator(File indexDir, String columnName, String[] sortedEntries)
+  public LuceneFSTIndexCreator(File indexDir, String columnName, String tableNameWithType, String[] sortedEntries)
       throws IOException {
+    this(indexDir, columnName, tableNameWithType, sortedEntries, new FSTBuilder());
+  }
+
+  @VisibleForTesting
+  public LuceneFSTIndexCreator(File indexDir, String columnName, String tableNameWithType, String[] sortedEntries,
+      FSTBuilder fstBuilder)
+      throws IOException {
+    _tableNameWithType = tableNameWithType;
     _fstIndexFile = new File(indexDir, columnName + V1Constants.Indexes.LUCENE_V912_FST_INDEX_FILE_EXTENSION);
 
-    _fstBuilder = new FSTBuilder();
+    _fstBuilder = fstBuilder;
     _dictId = 0;
     if (sortedEntries != null) {
       for (_dictId = 0; _dictId < sortedEntries.length; _dictId++) {
-        _fstBuilder.addEntry(sortedEntries[_dictId], _dictId);
+        try {
+          _fstBuilder.addEntry(sortedEntries[_dictId], _dictId);
+        } catch (IOException ex) {
+          // Caught exception while trying to add, update metric and skip the document
+          String metricKeyName =
+              _tableNameWithType + "-" + FstIndexType.INDEX_DISPLAY_NAME.toUpperCase(Locale.US) + "-indexingError";
+          ServerMetrics.get().addMeteredTableValue(metricKeyName, ServerMeter.INDEXING_FAILURES, 1);
```

Comment:
I think this should be optional, similar to how we do it with JSON indexes. Even if the default value for the option is to not fail, we should give users the ability to fail when invalid values are received

Reply:
Sure, I'll add this as a follow up PR